x-pack/plugins/enterprise_search/public/applications/app_search/components/document_creation/creation_mode_components/upload_json_file.tsx

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { useValues, useActions } from 'kea';
+
+import { i18n } from '@kbn/i18n';
+import {
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiButton,
+  EuiButtonEmpty,
+  EuiFilePicker,
+  EuiSpacer,
+  EuiText,
+} from '@elastic/eui';
+
+import { AppLogic } from '../../../app_logic';
+import { ConfiguredLimits } from '../../../types';
+
+import { MODAL_CANCEL_BUTTON, MODAL_CONTINUE_BUTTON } from '../constants';
+import { DocumentCreationLogic } from '../';
+
+export const UploadJsonFile: React.FC = () => (
+  <>
+    <ModalHeader />
+    <ModalBody />
+    <ModalFooter />
+  </>
+);
```

Comment:
This is super clean and easy to follow. I know you say the testing was harder but I feel like the tests might even be even more clear since everything is split up like you have it. And it's cool that you can just render the particular part that you want to test, rather than rendering the entire thing and then diving into the section you're interested in.